Source/Core/Rectangle.js

```diff
@@ -872,6 +874,198 @@ define([
         return result;
     };
 
+    var unprojectedScratch = new Cartographic();
+    var cornerScratch = new Cartographic();
+    var projectedScratch = new Cartesian3();
+    /**
+     * Approximates a Cartographic rectangle's extents in some map projection by projecting
+     * points in a grid throughout the rectangle.
+     *
+     * @function
+     *
+     * @param {Rectangle} cartographicRectangle An input rectangle in geographic coordinates.
+     * @param {MapProjection} mapProjection A MapProjection indicating a projection from geographic coordinates.
+     * @param {Rectangle} [result] Rectangle on which to store the projected extents of the input.
+     * @param {Number} [steps=8] Number of points to sample along each side of the geographic Rectangle.
+     */
+    Rectangle.approximateProjectedExtents = function(cartographicRectangle, mapProjection, result, steps) {
```

Comment:
2 optional parameters is bad API design and gets worse when result parameters are involved.  Have the first parameter be an `options` object and the second parameter be the `result`.  Also, does result need to be optional?